### PR TITLE
chore: Add Design process description & basic style guide

### PR DESCRIPTION
**Pull request by RomainMuller**
Fixes awslabs/cdk-ops#177

----

### Pull Request Checklist

Please check all boxes (including N/A items)

#### Testing

- [x] Unit test and/or integration test added
- [x] __Toolkit change?:__ [integration
tests](https://github.com/awslabs/aws-cdk/blob/master/packages/aws-cdk/integ-tests/test.sh)
manually executed (paste output to the PR description)
- [x] __Init template change?:__ coordinated update of integration tests
(currently maintained in a private repo).

#### Documentation

- [x] __README__: README and/or documentation topic updated
- [x] __jsdocs__: All public APIs documented

### Title and description

- [x] __Change type__: Title is prefixed with change type:
 * `fix(module): <title>` bug fix (_patch_)
 * `feat(module): <title>` feature/capability (_minor_)
 * `chore(module): <title> ` won't appear in changelog
 * `build(module): <title>` won't appear in changelog
- [x] __Title format__: Title uses lower case and doesn't end with a period
- [x] __Breaking change?__: Last paragraph of description is: `BREAKING
  CHANGE: <describe exactly what changed and how to achieve similar behavior +
  link to documentation/gist/issue if more details are required>`
- [x] __References__: Indicate issues fixed via: `Fixes #xxx` or `Closes #xxx`

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -41,6 +41,28 @@ and let us know if it's not up-to-date (even better, submit a PR with your  corr
 7. Once approved and tested, a maintainer will squash-merge to master and will use your PR title/description as the
    commit message.
 
+## Design Process
+
+In order to enable efficient collaboration over design documents, the following process must be followed:
+
+1. Open an issue describing the requirements and constraints the design must satisfy
+   + Provide a clear list of use-cases that the design intends to address
+2. Open a pull request with a Markdown document describing the proposed design
```
[the remark this answers is not in the document]
Comment (rix0rrr) [reply CONTRIBUTING.md]: No, but you know what? I think I'd like for the file to be committed along with the change.

Why not be the first project where we actually keep a set of design documents in our repo, as use to future historians (and team members)?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -41,6 +41,28 @@ and let us know if it's not up-to-date (even better, submit a PR with your  corr
 7. Once approved and tested, a maintainer will squash-merge to master and will use your PR title/description as the
    commit message.
 
+## Design Process
+
+In order to enable efficient collaboration over design documents, the following process must be followed:
```
Comment (eladb) [inline CONTRIBUTING.md]: "must" is too strong a word. I would say "recommended"

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (eladb) [reply CONTRIBUTING.md]: I would create a directory `/design` and recommend that people put the files there.

[the remark this answers is not in the document]
Comment (RomainMuller) [reply CONTRIBUTING.md]: @rix0rrr I **VERY** definitely meant for the documents to live with the project, I just meant we don't have the folder yet... But yeah I can recommend a file name and path.

**Pull request merged**